c/tskit/trees.c

```diff
@@ -4322,7 +4322,7 @@ tsk_diff_iter_init(tsk_diff_iter_t *self, tsk_treeseq_t *tree_sequence)
     self->removal_index = 0;
     self->tree_left = 0;
     self->tree_index = (size_t) -1;
-    self->edge_list_nodes = malloc(self->num_edges * sizeof(tsk_edge_list_t));
+    self->edge_list_nodes = malloc(self->num_edges * sizeof(tsk_edge_list_node_t));
```

Comment:
Best to change these to ``sizeof(*self->edge_list_nodes)`` as we're updating them - it's much more robust.